src/audio/dai-zephyr.c

```diff
@@ -1106,14 +1106,13 @@ static int dai_comp_trigger_internal(struct comp_dev *dev, int cmd)
 		break;
 	case COMP_TRIGGER_PAUSE:
 		comp_dbg(dev, "dai_comp_trigger_internal(), PAUSE");
+		dai_trigger_op(dd->dai, cmd, dev->direction);
 		if (prev_state == COMP_STATE_ACTIVE) {
 			ret = dma_suspend(dd->chan->dma->z_dev, dd->chan->index);
 		} else {
 			comp_warn(dev, "dma was stopped earlier");
 			ret = 0;
 		}
-
-		dai_trigger_op(dd->dai, cmd, dev->direction);
```

Comment:
Should not be handled like the STOP? there is this comment in that section:
```
/*
 * Some platforms cannot just simple disable
 * DMA channel during the transfer,
 * because it will hang the whole DMA controller.
 * Therefore, stop the DMA first and let the DAI
 * drain the FIFO in order to stop the channel
 * as soon as possible.
 */
```

Reply:
@dbaluta did NXP have this issue with ordering or am I thinking about another platform. If so, may need a Kconfig.

Reply:
I remember having some issues on the NXP side with the ordering indeed. Kconfig sounds good.

Reply:
@lgirdwood  we did have this problem. See @iuliana-prodan  comment. I think we should create a special CONFIG_ENABLE_DMA_FIRST and put everything until this.

Reply:
@ranj063 yes, that's ok for i.MX.
Thank you!

Reply:
It's ok now @ranj063 
Thank you!